### DISCUSSION TITLE
Plot take over 10 hours to complete - How to optimize?

### DISCUSSION THREAD

**Discussion by EddyXX** [swar/Swar-Chia-Plot-Manager#547] [Q&A]
I am running Swar Manager parallel plotting 11-12 plots.
My rig:  AMD 3900x (12c/24t) 64g ram 3600mhz, 2 x 2t 980pro nvme (raid 0),

This is my config:
  - name: plot
    max_plots: 200
    farmer_public_key:
    pool_public_key:
    temporary_directory: E:\ChiaTemp1
    destination_directory: C:\Users\Eddie\Documents\Chia Stagger
    size: 32
    bitfield: true
    threads: 4
    buckets: 128
    memory_buffer: 4000
    max_concurrent: 12
    max_concurrent_with_start_early: 13
    stagger_minutes: 30
    max_for_phase_1: 4
    concurrency_start_early_phase: 4
    concurrency_start_early_phase_delay: 0
    temporary2_destination_sync: false

I even used a stagger ssd. Time for all phases is like this: 
Phase 1  /  Phase2  /  Phase3  /  Phase4
03:38:16 / 01:39:53 / 03:42:11 / 00:21:44

Any idea how to improve the performance? I would be really appreciate.

**Comment by Lyushen** [comment]
There is clearly cpu bottleneck, my (10850k) phase 1 is about 2.2h, give it more cores and lower a max_for_phase_1 to amount that your cpu can handle phase one and other phases on background without critically overloaded, 100% fine if you on full cycle. Look at processor time in the task manager, it shows you how long the task take to be done, it should be lower as possible at 100% load. 
And another advice, since you using windows the raid is not a best option, separate them and add one more task for another NVMe, it will double your IO, as I know windows can't use full potential of IO in raid.

**Reply by EddyXX** [reply]
Forward propagation table time: 2144.764 seconds. CPU (159.000%) Wed May 19 07:01:00 2021
this is one of the log for Phase 1
	First computation pass time: 1823.650 seconds. CPU (78.060%) Wed May 19 14:26:58 2021
this is Phase 3
Do you mean the CPU usage for Phase 1 should be lower than 100%?
And by adding one more task to another NVME, do you mean open 2 Swar manager in 2 separate CLI/PowerShell?

**Reply by Lyushen** [reply]
1. U can see the persantage of using cpu, but if you bottlenecking it CPU can't use more so it shows you as 79% but it actually could use more if there is free resources left... My using more than 250% per that phase, focus on get this persantage as higher as possible.

2. No, you can do it through config, one moment I will copy paste it

**Reply by Lyushen** [reply]
```
manager:
check_interval: 60
log_level: ERROR

global:
max_concurrent: 12

jobs:
- name: NVMe 1
max_plots: 999
farmer_public_key:
pool_public_key:
temporary_directory: D:\
temporary2_directory:
destination_directory: G:\
size: 32
bitfield: true
threads: 4
buckets: 128
memory_buffer: 4000
max_concurrent: 6
max_concurrent_with_start_early: 6
stagger_minutes: 60
max_for_phase_1: 3
concurrency_start_early_phase: 4
concurrency_start_early_phase_delay: 30
temporary2_destination_sync: false

- name: NVMe 2
max_plots: 999
farmer_public_key:
pool_public_key:
temporary_directory: E:\
temporary2_directory:
destination_directory: G:\
size: 32
bitfield: true
threads: 3
buckets: 128
memory_buffer: 4000
max_concurrent: 6
max_concurrent_with_start_early: 6
stagger_minutes: 60
max_for_phase_1: 3
concurrency_start_early_phase: 4
concurrency_start_early_phase_delay: 30
temporary2_destination_sync: false
```

**Reply by Lyushen** [reply]
The values are not for your cpu, config it, but that is how it works

**Reply by EddyXX** [reply]
Great, this is so clear I will try this configuration.
Thanks mate you saved my day.

**Reply by EddyXX** [reply]
One question, if both jobs run 3 plots in phase 1, there will be 6 plots using 4x3 + 3x3 = 21 thread. And there are also 6 plots in other phases. Won't it get the CPU overloaded? since my cpu has only 24 threads.

**Reply by Lyushen** [reply]
Yes, you are correct, it left only 3 threats for other phases. But it intentional overload to fillup the processor wait-list. As you saw in taskmanager it going up and down, so little bit overload actually healthy, as I said before check processor time in the task manager (right click on graphs and show processor time) and track it, it should be about 20% when the core fully loaded, ofc if your processor can handle it, mine working on 5Ghz per core and it about 200+% core in logs per phase one. Analyze this values, it simple math) No mate, have a great day, sir!

**Reply by EddyXX** [reply]
![1](https://user-images.githubusercontent.com/22362072/118768015-d532b080-b8c1-11eb-8e17-ef16b5205048.PNG)
This is graph from task manager when I run 10 plots, 4 in phase 1 with 4 threads.
Seems like CPU is far from fully utilised.

**Reply by Lyushen** [reply]
Looks like it is, than something else could be bottleneck. But you got the theory about cpu, it not less important. I can guess it your raid, it just not using cpu because your raid 100% loaded (even if it not showing, the statistics for raid is wrong in the task manager), separate them, then it will utilize your process cores

**Reply by Lyushen** [reply]
Mine looks like this and it not 100% all time because there unfilled background phases (2-5).this is only 5 phase1 and 4 backgrounds, and as you can see how filled it even without full task list

![o4iFVwM](https://user-images.githubusercontent.com/78591028/118770570-2a52d000-b88a-11eb-9273-207353c573e2.png)

**Reply by EddyXX** [reply]
Yeah I think I should squeeze out more power from my CPU. How many thread do you set for phase 1?
I can see you overclock to 4.97ghz, hows you cpu temperature? Mine is around 60 ℃ without any OC. Don't know if my 360 all-in-one water cooling can hanlde overclocking.

**Reply by EddyXX** [reply]
![2](https://user-images.githubusercontent.com/22362072/118771524-feedd680-b8c5-11eb-8c12-381e2967e4aa.PNG)

**Reply by Lyushen** [reply]
I set to 3, but I wanna do more tests with 4, 3 it because of golden middle of 250% per phase one.
It actually good, it's around 75-90 (78 avg), but it's okay for that processor, thermal limit here is 115 degrees, but locked on 100 for safety. And it's not even 360, just cheep 240

**Reply by EddyXX** [reply]
Hey mate, I am seeing a weird thing. I started 2 job, one of them using 4 threads, one using 3. The plot with 3 threads is faster than 4 threads. Do you have any clue why this happening?

**Reply by EddyXX** [reply]
The 3 thread plot utilized 250% CPU while the 4 thread plot utilized only 180%

**Reply by EddyXX** [reply]
Never mind. After 15mins 4 threads is definitely faster than 3 threads.

**Reply by Lyushen** [reply]
Yes yes, it what I know as a fact now, but it depends of systems. Actually the app is very poorly coded and it cannot use multithreading well. I have different effect, I don't know if I said it before, but in my rig 4 threads better than 3. So probably I'll use that number. This is try-fail field) A lot of questions and there is no one best config, because how chia.exe runs on different system) We only can use our great tool in our heads to make run it perfectly smooth) Enjoy of the process, that what matter)

**Comment by tomasbruzza** [comment]
Hey! I have the exact same problem. How many plots per day are you making? Do you found any solution to the +10 hour plot? I'm thinking in reducing the max on the phase 1 to 2 plots and the max plots in simultaneous to 6. I don't know why with this kind of processor (12C/24T) we are getting this low performance.
